### PR TITLE
Update dependencies to match package.json/Gemfile

### PR DESCRIPTION
**Pull request by dLobatog**
Also removed the old `nodejs-stats-webpack-plugin` leftovers from comps and props

### CODE REVIEW

**Comment by dLobatog** [comment]
Looks like babel-loader cannot build currently, locally I got:
https://gist.github.com/777c9d4b4a7385d659270d69b4f5d8f0

@mmoll I saw http://koji.katello.org/koji/buildinfo?buildID=16303 , you built it manually there or do you know if anything changed since then..?

**Comment by mmoll** [comment]
I used `tito release koji-foreman` and hope everything will be OK with that. ;)

**Comment by ehelms** [comment]
I think this is a result of how we migrated koji. I had similar issues with other packages. I'll re-test this once I fix it.

File: foreman/foreman.spec [hunk 1]
```diff
@@ -236,8 +236,8 @@ BuildRequires: npm(diff) >= 3.0.0
 BuildRequires: npm(diff) < 3.1.0
 BuildRequires: npm(expose-loader) < 0.7.0
 BuildRequires: npm(expose-loader) >= 0.6.0
-BuildRequires: npm(extract-text-webpack-plugin) < 3.0.0
-BuildRequires: npm(extract-text-webpack-plugin) >= 2.1.0
+BuildRequires: npm(extract-text-webpack-plugin) < 4.0.0
+BuildRequires: npm(extract-text-webpack-plugin) >= 3.0.0
```
Comment (ehelms) [inline foreman/foreman.spec]: This package is only at version 2.1.2 -- http://koji.katello.org/koji/packageinfo?packageID=713

**Comment by ehelms** [comment]
[test]

File: foreman/foreman.spec (hunk 1 above)
Comment (dLobatog) [inline foreman/foreman.spec]: I'll update it as it's `"extract-text-webpack-plugin": "^3.0.0"` on package.json

**Comment by dLobatog** [comment]
Now sprockets is > 3.0, will update that in a moment

File: foreman/foreman.spec [hunk 2]
```diff
@@ -183,8 +183,8 @@ BuildRequires: %{?scl_prefix}rubygem(activerecord-session_store) >= 0.1.1
 BuildRequires: %{?scl_prefix}rubygem(activerecord-session_store) < 2
 BuildRequires: %{?scl_prefix_ror}rubygem(sprockets) >= 3
 BuildRequires: %{?scl_prefix_ror}rubygem(sprockets) < 4
-BuildRequires: %{?scl_prefix_ror}rubygem(sprockets-rails) >= 2.3.3
-BuildRequires: %{?scl_prefix_ror}rubygem(sprockets-rails) < 3
+BuildRequires: %{?scl_prefix}rubygem(sprockets-rails) >= 3
+BuildRequires: %{?scl_prefix}rubygem(sprockets-rails) < 4
```
Comment (dLobatog) [inline foreman/foreman.spec]: We have to build this again in `tfm` SCL as  ror42 does not contain this package.

**Comment by ehelms** [comment]
[test]

**Comment by dLobatog** [comment]
Odd, ["Error: No Package found for tfm-rubygem(sprockets-rails) >= 3"](http://koji.katello.org/kojifiles/work/tasks/6631/16631/mock_output.log) , let's see...

**Comment by dLobatog** [comment]
Oh, http://koji.katello.org/koji/buildinfo?buildID=16374 hasn't finished yet. I will kick tests again when it finishes building.

**Comment by mmoll** [comment]
[test]

**Comment by dLobatog** [comment]
@mmoll @ehelms https://github.com/theforeman/foreman-packaging/pull/1727 needs to be merged first, 

```

+ NODE_ENV=production
+ webpack.js --bail --config config/webpack.config.js
child_process.js:524
    throw err;
    ^
Error: Command failed: /builddir/build/BUILD/foreman-1.16.0/script/plugin_webpack_directories.rb
    at checkExecSyncError (child_process.js:481:13)
    at execSync (child_process.js:521:13)
    at Object.<anonymous> (/builddir/build/BUILD/foreman-1.16.0/config/webpack.config.js:27:19)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
error: Bad exit status from /var/tmp/rpm-tmp.apgNKK (%build)
RPM build errors:
```

**Comment by ehelms** [comment]
@dLobatog want to just include it here so we get a final green PR?

**Comment by dLobatog** [comment]
@ehelms alrighty

**Comment by dLobatog** [comment]
New error - I'll check what's going on http://koji.katello.org/kojifiles/work/tasks/6666/16666/build.log

**Comment by ehelms** [comment]
Looks like we need to package YAM? (yet another module)

**Comment by dLobatog** [comment]
@ehelms I don't think so, `nodejs-react` should already provide `object-assign` but it can't find it

**Comment by ehelms** [comment]
Did you mean to push Foreman 1.16 binary?

**Comment by dLobatog** [comment]
@ehelms Nope, I was just doing local tests and pushed accidentally. The result is the same, react can't read object-assign.

**Comment by dLobatog** [comment]
Locally I'm hitting https://bugzilla.redhat.com/show_bug.cgi?id=1457763 and I think we may be hitting the same, http-parser isn't linkable if it's coming from nodejs010 (the SCL). Koji seems to be installing nodejs010-http-parser - I made a small change to temporarily require http-parser (not from SCL) to see if it works (locally it fixed it)

**Comment by dLobatog** [comment]
Same 😢

**Comment by dLobatog** [comment]
Finally got it to compile assets. We're missing many things:

* nodejs-react-numeric-input https://github.com/theforeman/foreman-packaging/pull/1750
* rubygem-dynflow https://github.com/theforeman/foreman-packaging/pull/1745
* rubygem-daemons 
* rubygem-get_process_mem (will submit a PR for these 2 in a few)
* nodejs-extract-text-webpack 3.0 is incompatible with nodejs-webpack 3.0, it should be a newer release. 3.4.1 works fine. https://github.com/theforeman/foreman-packaging/pull/1753
* The webpack config in Foreman core is wrong. By using `path.join` in the resolve.modules section, it's making Webpack NOT look into nested node_modules directories. I changed it to:
``` 
resolve: {
    modules: [
      path.join(__dirname, '..', 'webpack'),
      'node_modules/'
    ],
 ```
Which resolves nested node_modules, and compiles fine.

File: foreman/foreman.spec [hunk 3]
```diff
@@ -199,19 +203,24 @@ BuildRequires: %{?scl_prefix_ror}rubygem(mail) >= 2.6
 BuildRequires: %{?scl_prefix_ror}rubygem(mail) < 3.0
 BuildRequires: %{?scl_prefix}rubygem(sshkey) >= 1.9
 BuildRequires: %{?scl_prefix}rubygem(sshkey) < 2.0
+BuildRequires: %{?scl_prefix}rubygem(dynflow) >= 0.8.25
+BuildRequires: %{?scl_prefix}rubygem(dynflow) < 1.0
+BuildRequires: %{?scl_prefix}rubygem(daemons)
+BuildRequires: %{?scl_prefix}rubygem(get_process_mem)
```
Comment (ehelms) [inline foreman/foreman.spec]: Should these also be runtime requires?

**Comment by ehelms** [comment]
Dynflow update built, nodejs-react-numeric-input, and webpack update.

File: foreman/foreman.spec (hunk 3 above)
Comment (dLobatog) [inline foreman/foreman.spec]: Yup, look above https://github.com/theforeman/foreman-packaging/pull/1739/files#diff-67e0a6165416d559ed233a03cbf4dbc1R122

**Comment by ehelms** [comment]
I found this will need the following:

```
diff --git a/comps/comps-foreman-plugins-rhel7.xml b/comps/comps-foreman-plugins-rhel7.xml
index 31101bb..be182b7 100644
--- a/comps/comps-foreman-plugins-rhel7.xml
+++ b/comps/comps-foreman-plugins-rhel7.xml
@@ -44,17 +44,13 @@
       <packagereq type="default">rubygem-smart_proxy_pulp</packagereq>
       <packagereq type="default">rubygem-smart_proxy_remote_execution_ssh</packagereq>
       <packagereq type="default">rubygem-smart_proxy_salt</packagereq>
-      <packagereq type="default">tfm-rubygem-algebrick</packagereq>
       <packagereq type="default">tfm-rubygem-angular-rails-templates</packagereq>
-      <packagereq type="default">tfm-rubygem-apipie-params</packagereq>
       <packagereq type="default">tfm-rubygem-azure</packagereq>
       <packagereq type="default">tfm-rubygem-azure-core</packagereq>
       <packagereq type="default">tfm-rubygem-bastion</packagereq>
       <packagereq type="default">tfm-rubygem-bastion-devel</packagereq>
       <packagereq type="default">tfm-rubygem-bootstrap-datepicker-rails</packagereq>
       <packagereq type="default">tfm-rubygem-commonjs</packagereq>
-      <packagereq type="default">tfm-rubygem-concurrent-ruby</packagereq>
-      <packagereq type="default">tfm-rubygem-concurrent-ruby-edge</packagereq>
       <packagereq type="default">tfm-rubygem-deface</packagereq>
       <packagereq type="default">tfm-rubygem-diffy</packagereq>
       <packagereq type="default">tfm-rubygem-docker-api</packagereq>
@@ -162,16 +158,12 @@
       <packagereq type="default">rubygem-smart_proxy_pulp-doc</packagereq>
       <packagereq type="default">rubygem-smart_proxy_remote_execution_ssh-doc</packagereq>
       <packagereq type="default">rubygem-smart_proxy_salt-doc</packagereq>
-      <packagereq type="default">tfm-rubygem-algebrick-doc</packagereq>
       <packagereq type="default">tfm-rubygem-angular-rails-templates-doc</packagereq>
-      <packagereq type="default">tfm-rubygem-apipie-params-doc</packagereq>
       <packagereq type="default">tfm-rubygem-azure-core-doc</packagereq>
       <packagereq type="default">tfm-rubygem-azure-doc</packagereq>
       <packagereq type="default">tfm-rubygem-bastion-doc</packagereq>
       <packagereq type="default">tfm-rubygem-bootstrap-datepicker-rails-doc</packagereq>
       <packagereq type="default">tfm-rubygem-commonjs-doc</packagereq>
-      <packagereq type="default">tfm-rubygem-concurrent-ruby-doc</packagereq>
-      <packagereq type="default">tfm-rubygem-concurrent-ruby-edge-doc</packagereq>
       <packagereq type="default">tfm-rubygem-deface-doc</packagereq>
       <packagereq type="default">tfm-rubygem-diffy-doc</packagereq>
       <packagereq type="default">tfm-rubygem-docker-api-doc</packagereq>
diff --git a/comps/comps-foreman-rhel7.xml b/comps/comps-foreman-rhel7.xml
index e7dafc3..ca0f60b 100644
--- a/comps/comps-foreman-rhel7.xml
+++ b/comps/comps-foreman-rhel7.xml
@@ -39,8 +39,10 @@
       <packagereq type="default">tfm-mod_passenger</packagereq>
       <packagereq type="default">tfm-rubygem-activerecord-session_store</packagereq>
       <packagereq type="default">tfm-rubygem-addressable</packagereq>
+      <packagereq type="default">tfm-rubygem-algebrick</packagereq>
       <packagereq type="default">tfm-rubygem-ancestry</packagereq>
       <packagereq type="default">tfm-rubygem-apipie-bindings</packagereq>
+      <packagereq type="default">tfm-rubygem-apipie-params</packagereq>
       <packagereq type="default">tfm-rubygem-apipie-rails</packagereq>
       <packagereq type="default">tfm-rubygem-audited</packagereq>
       <packagereq type="default">tfm-rubygem-autoparse</packagereq>
@@ -49,6 +51,8 @@
       <packagereq type="default">tfm-rubygem-bootstrap-sass</packagereq>
       <packagereq type="default">tfm-rubygem-bundler_ext</packagereq>
       <packagereq type="default">tfm-rubygem-clamp</packagereq>
+      <packagereq type="default">tfm-rubygem-concurrent-ruby</packagereq>
+      <packagereq type="default">tfm-rubygem-concurrent-ruby-edge</packagereq>
       <packagereq type="default">tfm-rubygem-css_parser</packagereq>
       <packagereq type="default">tfm-rubygem-daemons</packagereq>
       <packagereq type="default">tfm-rubygem-deacon</packagereq>
@@ -276,8 +280,10 @@
       <packagereq type="default">rubygem-rubyipmi-doc</packagereq>
       <packagereq type="default">tfm-rubygem-activerecord-session_store-doc</packagereq>
       <packagereq type="default">tfm-rubygem-addressable-doc</packagereq>
+      <packagereq type="default">tfm-rubygem-algebrick-doc</packagereq>
       <packagereq type="default">tfm-rubygem-ancestry-doc</packagereq>
       <packagereq type="default">tfm-rubygem-apipie-bindings-doc</packagereq>
+      <packagereq type="default">tfm-rubygem-apipie-params-doc</packagereq>
       <packagereq type="default">tfm-rubygem-apipie-rails-doc</packagereq>
       <packagereq type="default">tfm-rubygem-audited-doc</packagereq>
       <packagereq type="default">tfm-rubygem-autoparse-doc</packagereq>
@@ -286,6 +292,8 @@
       <packagereq type="default">tfm-rubygem-bootstrap-sass-doc</packagereq>
       <packagereq type="default">tfm-rubygem-bundler_ext-doc</packagereq>
       <packagereq type="default">tfm-rubygem-clamp-doc</packagereq>
+      <packagereq type="default">tfm-rubygem-concurrent-ruby-doc</packagereq>
+      <packagereq type="default">tfm-rubygem-concurrent-ruby-edge-doc</packagereq>
       <packagereq type="default">tfm-rubygem-css_parser-doc</packagereq>
       <packagereq type="default">tfm-rubygem-daemons-doc</packagereq>
       <packagereq type="default">tfm-rubygem-deacon-doc</packagereq>
diff --git a/rel-eng/tito.props b/rel-eng/tito.props
index 623158b..f02df7e 100644
--- a/rel-eng/tito.props
+++ b/rel-eng/tito.props
@@ -94,10 +94,12 @@ whitelist = foreman-bootloaders-redhat
 [foreman-nightly-rhel7]
 disttag = .el7
 whitelist = foreman
+  rubygem-algebrick
   rubygem-activerecord-session_store
   rubygem-addressable
   rubygem-ancestry
   rubygem-apipie-bindings
+  rubygem-apipie-params
   rubygem-apipie-rails
   rubygem-audited
   rubygem-autoparse
@@ -105,12 +107,14 @@ whitelist = foreman
   rubygem-awesome_print
   rubygem-bootstrap-sass
   rubygem-bundler_ext
+  rubygem-concurrent-ruby-edge
   rubygem-clamp
   rubygem-css_parser
   rubygem-daemons
   rubygem-deacon
   rubygem-deep_cloneable
   rubygem-domain_name
+  rubygem-dynflow
   rubygem-eventmachine
   rubygem-excon
   rubygem-extlib
@@ -132,6 +136,7 @@ whitelist = foreman
   rubygem-font-awesome-sass
   rubygem-formatador
   rubygem-friendly_id
+  rubygem-get_process_mem
   rubygem-gettext
   rubygem-gettext_i18n_rails
   rubygem-gettext_i18n_rails_js
@@ -212,20 +217,16 @@ scl = tfm
 [foreman-plugins-nightly-rhel7]
 disttag = .el7
 scl = tfm
-whitelist = rubygem-algebrick
-  rubygem-angular-rails-templates
-  rubygem-apipie-params
+whitelist = rubygem-angular-rails-templates
   rubygem-azure
   rubygem-azure-core
   rubygem-bastion
   rubygem-bootstrap-datepicker-rails
   rubygem-commonjs
   rubygem-concurrent-ruby
-  rubygem-concurrent-ruby-edge
   rubygem-deface
   rubygem-diffy
   rubygem-docker-api
-  rubygem-dynflow
   rubygem-faraday_middleware
   rubygem-ffi
   rubygem-fog-azure
@@ -270,7 +271,6 @@ whitelist = rubygem-algebrick
   rubygem-foreman_vmwareannotations
   rubygem-foreman_xen
   rubygem-foreman_templates
-  rubygem-get_process_mem
   rubygem-graphite-api
   rubygem-hammer_cli_foreman_admin
   rubygem-hammer_cli_foreman_bootdisk
```

**Comment by ehelms** [comment]
Built fine after that -- http://koji.katello.org/koji/taskinfo?taskID=18120

**Comment by dLobatog** [comment]
[test]

**Comment by ehelms** [comment]
Thanks @dLobatog !

**Pull request merged**